### PR TITLE
CORDA-680 Update cordapp packages documentation

### PR DESCRIPTION
**Pull request by andr3ej**
and introduce MockNetworkParameters.

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 1]
```diff
@@ -62,6 +62,23 @@ fun StartedNode<MockNetwork.MockNode>.pumpReceive(block: Boolean = false): InMem
...
+        val defaultFactory: MockNetwork.Factory<*> = MockNetwork.DefaultFactory,
+        val initialiseSerialization: Boolean = true,
+        val cordappPackages: List<String> = emptyList()) {
+    fun setNetworkSendManuallyPumped(networkSendManuallyPumped: Boolean) = copy(networkSendManuallyPumped = networkSendManuallyPumped)
+    fun setThreadPerNode(threadPerNode: Boolean) = copy(threadPerNode = threadPerNode)
+    fun setServicePeerAllocationStrategy(servicePeerAllocationStrategy: InMemoryMessagingNetwork.ServicePeerAllocationStrategy) = copy(servicePeerAllocationStrategy = servicePeerAllocationStrategy)
+    fun setDefaultFactory(defaultFactory: MockNetwork.Factory<*>) = copy(defaultFactory = defaultFactory)
+    fun setInitialiseSerialization(initialiseSerialization: Boolean) = copy(initialiseSerialization = initialiseSerialization)
+    fun setCordappPackages(cordappPackages: List<String>) = copy(cordappPackages = cordappPackages)
+}
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: similar classes exist for driver

File: docs/source/upgrade-notes.rst [hunk 2]
```diff
@@ -182,10 +182,9 @@ We will be revisiting this capability in a future release.
 
 * CorDapps must be explicitly registered in ``MockNetwork`` unit tests.
 
-  This is done by calling ``setCordappPackages``, an extension helper function in the ``net.corda.testing`` package,
-  on the first line of your `@Before` method. This takes a variable number of `String` arguments which should be the
+  This is done via the ``cordappPackages`` constructor parameter of MockNetwork.
+  This takes a list of `String` values which should be the
   package names of the CorDapps containing the contract verification code you wish to load.
-  You should unset CorDapp packages in your `@After` method by using ``unsetCordappPackages()`` after `stopNodes()`.
```
Comment (andr3ej) [inline docs/source/upgrade-notes.rst]: should these changes go in a 1.1 section?

Comment (josecoll) [reply docs/source/upgrade-notes.rst]: Yes!

Comment (andr3ej) [reply docs/source/upgrade-notes.rst]: Done

File: docs/source/upgrade-notes.rst [hunk 3]
```diff
@@ -25,6 +25,19 @@ versions you are currently using are still in force.
...
+
+Testing
+^^^^^^^
+
+* The registration mechanism for CorDapps in ``MockNetwork`` unit tests has changed.
+
+  It is now done via the ``cordappPackages`` constructor parameter of MockNetwork.
+  This takes a list of `String` values which should be the
+  package names of the CorDapps containing the contract verification code you wish to load.
+  The ``unsetCordappPackages`` method is now redundant and has been removed.
```
Comment (Clintonio) [inline docs/source/upgrade-notes.rst]: Yay!

File: docs/source/upgrade-notes.rst (hunk 2 above)
Comment (Clintonio) [reply docs/source/upgrade-notes.rst]: Isn't this for 2.0? I thought 1.1 was 0 API additions or changes.

Comment (andr3ej) [reply docs/source/upgrade-notes.rst]: according to the jira ticket this comes under test-utils stability, which mike has made my top priority. the work is already merged so let's work it out at release time - i've put the upgrade notes in an "unreleased" section for now.

**Pull request merged**